app/api/check_cumulative.py

```diff
@@ -0,0 +1,109 @@
+"""
+Check for cumulative data decreases from week to week.
+"""
+import csv
+from time import time
+
+import flask
+import pandas as pd
+
+from app.api import utils
+
+def do_check_cumulative(outputFile):
+    states = utils.get_all_state_finals()
+
+    with open(outputFile, mode='w') as csv_file:
+        fieldnames = ['State', 'Facility', 'Date', 'cume_res_pos_value', 'prev_cume_res_pos_value']
+        writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+        writer.writeheader()
+
+        def check_cumulative(df):
+            collection_dates = df[['Date']].drop_duplicates()
+
+            last_cumes = {}
+            last_cume = 0
+
+            for date_index, date_row in collection_dates.iterrows():
+                collection_date = date_row['Date']
+                current_block = df.loc[df['Date'] == collection_date]
+
+                for _, f_row in current_block.iterrows():
+                    current_cume = f_row['Cume_Res_Pos']
```

Comment:
I think this is assuming that the dates are in chronological order as you go through them, is that guaranteed?

Also for future reference, Pandas has nifty ways of making this kind of stuff easier. Ultimately you're taking two columns and calculating the difference between each row and the one before it, and you can do that with the `diff` function (see e.g. https://stackoverflow.com/questions/60224216/how-to-calculate-change-between-rows-in-pandas-data-frame) and then filter to cases where that difference is negative.

Reply:
@julia326 its not guaranteed which indeed caused issues. Do you have any easy way to check that?

Reply:
ah perfect - thank you!